frontend/packages/dev-console/src/utils/__tests__/pipeline-utils.spec.ts

```diff
@@ -56,4 +63,22 @@ describe('pipeline-utils ', () => {
     expect(data.status).not.toBeNull();
     expect(data.status).toBe('Pending');
   });
+
+  it('expect duration to be "-" for PipelineRun without start Time', () => {
+    const duration = pipelineRunDuration(mockRunDurationTest[0]);
+    expect(duration).not.toBeNull();
+    expect(duration).toBe('-');
+  });
+
+  it('expect duration to be "-" for non running PipelineRun without end Time', () => {
+    const duration = pipelineRunDuration(mockRunDurationTest[1]);
+    expect(duration).not.toBeNull();
+    expect(duration).toBe('-');
+  });
+
+  it('expect duration to be a time formatted string for PipelineRun with start and end Time', () => {
+    const duration = pipelineRunDuration(mockRunDurationTest[2]);
+    expect(duration).not.toBeNull();
+    expect(duration).not.toBe('-');
+  });
 });
```

Comment:
Please add success path tests as well. If the tests break at a point in the future, it is nice to know if the whole thing just failed or if just a null-path failed.

Reply:
This is a success test. This checks for the value to be not `-`. i.e some time string.
`expect(duration).not.toBe('-');`
I am updating this to the exact time-string now.

Reply:
Apparently I am blind... did not see `not`. The test is pretty open-ended, and doesn't cover the values of success. Perhaps we should still write tests to test known success cases.

Reply:
Apparently I am double blind lol - you said `I am updating this to the exact time-string now`. Gosh, that's what I get for starting my day after a long weekend out coffee ;)

Reply:
Haha, never mind :)